patchwork-extensions-block/src/main/java/net/patchworkmc/mixin/extensions/block/harvest/MixinServerPlayerInteractionManager.java

```diff
@@ -0,0 +1,151 @@
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.network.ServerPlayerInteractionManager;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.GameMode;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.World;
+
+import net.patchworkmc.impl.extensions.block.BlockContext;
+import net.patchworkmc.impl.extensions.block.BlockHarvestManager;
+import net.patchworkmc.impl.extensions.block.Signatures;
+
+@Mixin(ServerPlayerInteractionManager.class)
+public abstract class MixinServerPlayerInteractionManager {
+	// removedByPlayer, canHarvestBlock
+
+	@Unique
+	private boolean patchwork$removeBlock(BlockPos pos, boolean canHarvest) {
+		ServerPlayerInteractionManager me = (ServerPlayerInteractionManager) (Object) this;
+		BlockState state = me.world.getBlockState(pos);
+		boolean removed = ((IForgeBlockState) state).removedByPlayer(me.world, pos, me.player, canHarvest, me.world.getFluidState(pos));
+
+		if (removed) {
+			state.getBlock().onBroken(me.world, pos, state);
+		}
+
+		return removed;
+	}
+
+	/////////////////////////////////////
+	/// Fire BlockEvent.BreakEvent,
+	/// get the amount of exp to drop.
+	/////////////////////////////////////
+	@Shadow
+	public ServerWorld world;
+	@Shadow
+	public ServerPlayerEntity player;
+	@Shadow
+	private GameMode gameMode;
+
+	@Inject(method = "tryBreakBlock", at = @At("HEAD"), cancellable = true)
+	private void hookBreakBlock(BlockPos pos, CallbackInfoReturnable<Boolean> callback) {
+		int exp = BlockHarvestManager.onBlockBreakEvent(world, gameMode, player, pos);
+
+		if (exp < 0) {
+			callback.setReturnValue(false);
+		} else {
+			BlockHarvestManager.pushExpDropStack(exp);
+		}
+	}
+
+	/////////////////////////////////////
+	@Redirect(method = "tryBreakBlock", at = @At(value = "INVOKE", target = Signatures.Block_onBreak, ordinal = 0))
+	private void patchwork$tryBreakBlock_onBreak(Block block, World world, BlockPos pos, BlockState state, PlayerEntity player) {
+		// Suppress this call
+	}
+
+	@Redirect(method = "tryBreakBlock", at = @At(value = "INVOKE", target = Signatures.ServerWorld_removeBlock, ordinal = 0))
+	private boolean patchwork$tryBreakBlock_removeBlock(ServerWorld world, BlockPos pos, boolean bool) {
+		return true; // bypass if (bl && bl2) {
+	}
+
+	@Redirect(method = "tryBreakBlock", at = @At(value = "INVOKE", target = Signatures.Block_onBroken, ordinal = 0))
+	private void patchwork$tryBreakBlock_onBroken(Block block, IWorld world, BlockPos pos, BlockState state) {
+		// Suppress this call
+	}
+
+	@Redirect(method = "tryBreakBlock", at = @At(value = "INVOKE", target = Signatures.ServerPlayerInteractionManager_isCreative, ordinal = 0))
```

Comment:
I don't think this needs to be a redirect. One option could be to @Slice from the isCreative check and inject at RETURN ordinal = 0.